routes/user.js

```diff
@@ -0,0 +1,73 @@
+var express = require("express");
+var app = express();
+const bcrypt = require("bcrypt");
+var { auth, jwtAuth } = require("../config/auth");
+var { User } = require("../models/User");
+const jwt = require("jsonwebtoken");
+
+function generateAccessToken(userId) {
+  return jwt.sign(userId, process.env.TOKEN_SECRET);
+}
+
+app.post("/register", async (req, res) => {
+  const salt = await bcrypt.genSalt();
+  const userPassword = await bcrypt.hash(req.body.password, salt);
+  if (req.body.password !== req.body.confirmpassword) {
+    res.status(500).json({
+      error: 1,
+      message: "password not matched",
+      data: null,
+    });
+  } else {
+    const userData = new User({
+      username: req.body.username,
+      dob: req.body.dob.toString(),
+      password: userPassword,
+      email: req.body.email,
```

Comment:
add hooks so that email should be unique using mongoose and username also

Reply:
added validation for username and email